lapis/src/main/kotlin/org/genspectrum/lapis/model/AdvancedQueryCustomListener.kt

```diff
@@ -0,0 +1,534 @@
+                try {
+                    expressionStack.addLast(IntEquals(field.name, metadataValue.toInt()))
+                } catch (e: NumberFormatException) {
+                    throw BadRequestException("'$metadataValue' is not a valid integer", e)
+                }
+            }
+
+            is SequenceFilterFieldType.IntFrom -> {
+                throw BadRequestException(
+                    "Cannot use $metadataName in advancedQuery, use >=(${field.name}) instead",
+                    null,
+                )
+            }
+
+            is SequenceFilterFieldType.IntTo -> {
+                throw BadRequestException(
+                    "Cannot use $metadataName in advancedQuery, use <=(${field.name}) instead",
+                    null,
+                )
+            }
+
+            is SequenceFilterFieldType.StringSearch -> {
+                expressionStack.addLast(StringSearch(field.type.associatedField, metadataValue))
+            }
+
+            SequenceFilterFieldType.VariantQuery -> {
+                throw BadRequestException("$VARIANT_QUERY_FIELD cannot be called from advanced query", null)
+            }
+
+            SequenceFilterFieldType.AdvancedQuery -> {
+                throw BadRequestException("$ADVANCED_QUERY_FIELD cannot be called recursively", null)
+            }
+        }
+    }
+
+    override fun enterRegexMetadataQuery(ctx: AdvancedQueryParser.RegexMetadataQueryContext) {
+        val metadataName = ctx.geneOrName().text
+        val metadataValue = ctx.value().text.trim('\'')
+
+        val field: SequenceFilterField? = allowedSequenceFilterFields.fields[metadataName.lowercase(Locale.US)]
+        field ?: throw BadRequestException("Metadata field $metadataName does not exist", null)
+        when (field.type) {
+            SequenceFilterFieldType.String -> {
+                expressionStack.addLast(StringEquals(field.name, metadataValue))
+            }
+
+            is SequenceFilterFieldType.StringSearch -> {
+                expressionStack.addLast(StringSearch(field.type.associatedField, metadataValue))
+            }
+
+            SequenceFilterFieldType.Lineage -> {
+                val lineage = mapToLineageFilter(
+                    field.name,
+                    metadataValue,
+                )
+                expressionStack.addLast(lineage)
+            }
+
+            SequenceFilterFieldType.VariantQuery -> {
+                throw BadRequestException("$VARIANT_QUERY_FIELD cannot be called from advanced query", null)
+            }
+
+            SequenceFilterFieldType.AdvancedQuery -> {
+                throw BadRequestException("$ADVANCED_QUERY_FIELD cannot be called recursively", null)
+            }
+
+            else -> {
+                throw BadRequestException(
+                    "Expression contains symbols not allowed for metadata field of type ${field.type} (allowed symbols: a-z, A-Z, 0-9, ., *, -, _)",
+                    null,
+                )
```

Comment:
The expression `intField = '1'` yields the error message "Expression contains symbols not allowed for metadata field of type Int (allowed symbols: a-z, A-Z, 0-9, ., *, -, _)" which is rather confusing IMO. I wouldn't really know what's wrong if I was a user.

-> Maybe we should simply implement the regex search as a parser rule instead? I.e.:
`regexMetadataQuery: geneOrName DOT R E G E X '=' value;`
That would save us from that problem. We would only enter the regex rule when the ".regex" is present in the field name.

Reply:
I will improve the message!

I also need to require exact string matches with spaces or other special characters to use the QUOTED_STRING param - so I think fixing that metadata fields end with regex is a possible solution

Reply:
Potentially this is also a naming issue - because I have to wrap exact searches with spaces they will also be wrapped with `' ' ` and end up in this case - so queries that are matched are not only regex.